zeebe/gateway-protocol/src/main/proto/v2/process-instances.yaml

```diff
@@ -0,0 +1,1291 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProcessInstanceElementStatisticsQueryResult'
+        "400":
+          $ref: 'common-responses.yaml#/components/responses/InvalidData'
+        "401":
+          $ref: 'common-responses.yaml#/components/responses/Unauthorized'
+        "403":
+          $ref: 'common-responses.yaml#/components/responses/Forbidden'
+        "500":
+          $ref: 'common-responses.yaml#/components/responses/InternalServerError'
+      x-eventually-consistent: true
+
+  /process-instances/search:
+    post:
+      tags:
+        - Process instance
+      operationId: searchProcessInstances
+      summary: Search process instances
+      description: Search for process instances based on given criteria.
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ProcessInstanceSearchQuery'
+      responses:
+        "200":
+          description: The process instance search result.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProcessInstanceSearchQueryResult'
+        "400":
+          $ref: 'common-responses.yaml#/components/responses/InvalidData'
+        "401":
+          $ref: 'common-responses.yaml#/components/responses/Unauthorized'
+        "403":
+          $ref: 'common-responses.yaml#/components/responses/Forbidden'
+        "500":
+          $ref: 'common-responses.yaml#/components/responses/InternalServerError'
+      x-eventually-consistent: true
+
+  /process-instances/{processInstanceKey}/incidents/search:
+    post:
+      tags:
+        - Process instance
+      operationId: searchProcessInstanceIncidents
+      summary: Search for incidents associated with a process instance
+      description: |
+        Search for incidents caused by the process instance or any of its called process or decision instances.
+
+        Although the `processInstanceKey` is provided as a path parameter to indicate the root process instance,
+        you may also include a `processInstanceKey` within the filter object to narrow results to specific
+        child process instances. This is useful, for example, if you want to isolate incidents associated with
+        subprocesses or called processes under the root instance while excluding incidents directly tied to the root.
+      parameters:
+        - name: processInstanceKey
+          in: path
+          required: true
+          description: The assigned key of the process instance, which acts as a unique identifier for this process instance.
+          schema:
+            $ref: 'keys.yaml#/components/schemas/ProcessInstanceKey'
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ProcessInstanceIncidentSearchQuery'
```

Comment:
❓ @nicpuppa, any idea why you've introduced this schema? It didn't exist in the original spec, and it's inconsistent with the original (missing the `filter` attribute).
💭 I would like to understand how this could happen and how we ensure this didn't also happen in other places. Effectively, we'd need to compare the old spec before the split with a consolidated/resolved version of the modular one 😅

Reply:
I think this is coming from the external specification. I hope we have test that covers the others. I'll adjust it tomorrow

Reply:
Oh yeah, I see. Interesting, what the f** happened there? 😆

Reply:
For the gateway code, it doesn't really matter because we don't generate the controller paths. Thus we can write and expect whatever we want there (still expecting the incident filter there as a request body). So, acceptance tests and client behavior is the same. But yeah, we don't detect this stuff then.